### PR TITLE
Report workflow failures to VK Teams

### LINKED ISSUE #3085: Fail-fast checking and workflow reporting in tarantool/doc
# Definition of done

- [x] Use strict error handling in bash scripts #3083
- [ ] Report failed workflows to VK Teams #3086

### PR DESCRIPTION
**Pull request by NickVolynkin**
Resolves #3085

### CODE REVIEW

File: .github/workflows/deploy-branch.yml [hunk 1]
```diff
@@ -14,38 +14,63 @@ jobs:
...
           submodules: recursive
 
       - name: Fetch submodules
+        id: fetch-submodules
         run: git fetch --recurse-submodules
 
       - name: Set branch name from source branch
-        run: echo "BRANCH_NAME=${GITHUB_HEAD_REF##*/}" >> $GITHUB_ENV
+        id: set-branch-name
+        run: echo "DEPLOYMENT_NAME=${GITHUB_HEAD_REF##*/}" >> $GITHUB_ENV
```
Comment (artembo) [inline .github/workflows/deploy-branch.yml:30]: BRANCH_NAME was much better, because it is a branch name. It looks like useless renaming, really.

**Review by lastoCHka42 — APPROVED** [review]

File: .github/workflows/destroy-deployment-manually.yml [hunk 2]
```diff
@@ -9,18 +9,41 @@ on:
...
+        id: delete-deployment-webhook
+        run: bash webhooks/delete_deployment.sh
+
+      - name: Remove GitHub deployment at ${{env.DEPLOYMENT_NAME}}
         uses: strumwolf/delete-deployment-environment@v2
+        id: remove-github-deployment
         with:
           token: "${{ secrets.TARANTOOLBOT_TOKEN }}"
           environment: ${{ env.DEPLOYMENT_NAME }}
 #          onlyRemoveDeployments: true
```
Comment (artembo) [inline .github/workflows/destroy-deployment-manually.yml]: please remove the comment

File: .github/workflows/destroy-deployment.yml [hunk 3]
```diff
@@ -1,35 +1,50 @@
...
+      VKTEAMS_BOT_TOKEN: ${{secrets.VKTEAMS_TARANTOOLBOT_TOKEN}}
     steps:
       - uses: actions/checkout@v2
+        id: checkout
         with:
           token: "${{ secrets.GITHUB_TOKEN }}"
 
       - name: Set branch name from source branch
-        run: echo "BRANCH_NAME=${GITHUB_HEAD_REF##*/}" >> $GITHUB_ENV
+        id: set-branch-name
```
Comment (artembo) [inline .github/workflows/destroy-deployment.yml]: why did you add so many id's? how are you going to use them?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -21,27 +21,44 @@ jobs:
...
+        id: make-json
       - run: make json-ru
+        id: make-json-ru
       - run: make pdf
+        id: make-pdf
       - run: make pdf-ru
+        id: make-pdf-ru
+
+      # set environment variable BRANCH_NAME, will be used in scripts
       - uses: nelonoel/branch-name@v1.0.1
```
Comment (artembo) [inline .github/workflows/main.yml]: DEPLOYMENT_NAME (ex. BRANCH_NAME) is a name of the current branch, you can use it instead of another external action

**Review by artembo — DISMISSED** [review]

File: .github/workflows/destroy-deployment.yml (hunk 3 above)
Comment (NickVolynkin) [reply .github/workflows/destroy-deployment.yml]: It's for tarantool/actions/report-job-status, see https://github.com/tarantool/actions/tree/master/report-job-status#prepare-your-workflow

File: .github/workflows/deploy-branch.yml (hunk 1 above)
Comment (NickVolynkin) [reply .github/workflows/deploy-branch.yml:30]: We're creating a deployment in this workflow, not a branch. The deployment's name can come from a branches name, from inputs or can be hardcoded for stable branches.

File: .github/workflows/destroy-deployment-manually.yml (hunk 2 above)
Comment (NickVolynkin) [reply .github/workflows/destroy-deployment-manually.yml]: done
